tests/test_biophysical/test_crop_soil_field/soil_tests/conftest.py

```
@@ -0,0 +1,39 @@
+happened to run the victim tests (which call the real methods) before the
+polluting ones, so the leak was invisible. pytest >= 8 changed collection
+ordering, exposing the leak: tests such as
+``test_layer_data.py::test_calculate_phosphorus_sorption_parameter`` then receive
+the leaked ``MagicMock`` instead of the real method.
+
+The autouse fixture below snapshots and restores ``LayerData``'s class state
+around every soil test, keeping such monkey-patches test-local without having to
+rewrite each call site. New leaks of the same kind are neutralised automatically.
+"""
+
+from typing import Iterator
```

Ruff (0.15.15)

[warning] Import from `collections.abc` instead: `Iterator`

Import from `collections.abc`

(UP035)